third-party/jemalloc/README

```diff
@@ -0,0 +1,11 @@
+==========================
+jemalloc README for Chapel
+==========================
+
+This copy of jemalloc is being released with Chapel for convenience and
+uses a snapshot from:
```

Comment:
For things like GASNet that are versioned, you can get the version number (implicit in the package dir hierarchy) and thus know what is in the package.  For things that are not versioned, like qthreads (currently) and now jemalloc, we need a way to do the same.  Qthreads is a bit of a special case in that we just pulled a tarball off their website.  So about the best we can do in terms of identifying exactly what we have is to look at the timestamp at which that code was added to our repo long ago and then ask the Qthreads folks what we must have gotten, given that timestamp.  For something where we're pulling from a repo we can and should do much better.  The timestamp on our commit that adds the jemalloc source tree is an okay(ish) identifier, but it still could be the case that mods could have gone into that repo between when we picked up the sources from their repo and when we committed them to our repo.  Better would be to record the jemalloc repo SHA representing the sources we picked up, when we put them into our repo.  Then if we have a question about what we have or don't have along their evolutionary path, we can always answer it accurately.